src/Common/src/CoreWCF/Runtime/ReentrantAsyncLock.cs

```diff
@@ -11,7 +11,7 @@
 
 namespace CoreWCF.Runtime
 {
-    internal class AsyncLock
+    internal class ReentrantAsyncLock
     {
```

Comment:
Can you replace the implementation of this class with the new implementation in dotnet/wcf as that one doesn't have the problems this one does.